janitor/functions.py

```diff
@@ -3236,6 +3236,49 @@ def groupby_agg(
     new_col = df.groupby(by)[agg_column_name].transform(agg)
     df_new = df.assign(**{new_column_name: new_col})
     return df_new
+    """
+    :Setup:
```

Comment:
I think this, and the `:Example:` and `:Output:` syntax are invalid Sphinx syntax. If it is indeed invalid, it was my fault for letting it through previously.

I actually like the example you provided, and I'm convinced we should include it, but it does contain a bunch of long  overhead that may hamper the readability of the doc page.

There's two ways I can think of to get around this:

1. Move the example into its own Jupyter notebook, and then link the Jupyter notebook instead (while also building the Jupyter notebooks as part of the docs).
2. Add a bit of prose under the first code block above, and then add in all of these code blocks you've added as a single code block with Python comments annotating what's happening in-line. 

As a maintainer, I would prefer the first, because it helps keep the design pattern where quick usage reminders are shown on the API docs, but details show up in a notebook (for which we can also use Binder to serve up). 

What do you say, @samukweku?

Reply:
thanks for the feedback; I'll go with the first option  - move the example to a jupyter notebook. Any examples in the docs where this has been done so I can model my approach on that?

Reply:
@samukweku awesomeness! 

To start, you'd put the notebooks in the [notebooks directory](https://github.com/ericmjl/pyjanitor/tree/dev/examples/notebooks).

I think that's it - we already have a bunch of infrastructure that lets us automatically pick out notebooks and build them.

To write a good notebook, make sure the purpose is written clearly, and that it has sufficient prose for others to understand. That's all we ask for :smile:.

Reply:
And btw, you can continue working on this branch to get it done, don't worry about creating a new one.

Reply:
@ericmjl 

made the requested changes - took out the example block and created a notebook for the groupby_agg example. Let me know your thoughts.

Reply:
@samukweku thanks for getting back! Have you pushed up the changes yet? They aren't yet reflected on the pull request. (If you're working on the `samukweku:myedits_pyjanitor` branch, and push any changes up to the same branch, they will magically be reflected on the pull request :tada:.)

Reply:
@ericmjl I think I got this time around. Small wins!